### DISCUSSION TITLE
clang-format indenting comment-only line incorrectly

### DISCUSSION THREAD

**Discussion by MatejVitek** [microsoft/vscode-cpptools#12300] [Q&A] [answered]
I have the following setup in my user settings.json:

	"C_Cpp.clang_format_style": "{BasedOnStyle: LLVM, UseTab: ForIndentation, IndentWidth: 4, TabWidth: 4, ColumnLimit: 0, PointerAlignment: Left, BreakBeforeBraces: Attach, SortIncludes: CaseInsensitive, AllowShortIfStatementsOnASingleLine: Never, AlignTrailingComments: Leave, SpacesBeforeTrailingComments: 2, SpaceBeforeCaseColon: false, InsertNewlineAtEOF: true}",
	"C_Cpp.clang_format_fallbackStyle": "Visual Studio",

However, this setup is indenting my comment lines incorrectly. My expected result would look like:

	int main(int argc, char* argv[]) {
		// create 10 threads
		pthread_t threads[10];
		for (int i = 0; i < 10; i++)
			pthread_create(&threads[i], NULL, thread, NULL);

		// wait for a bit then stop the threads
		sleep(3);
		for (int i = 0; i < 10; i++) {
			// pthread_yield();  // deprecated
			sched_yield();
			pthread_cancel(threads[i]);  // cancel thread
		}
	}

However, the result of formatting with the above setup is the following:

	int main(int argc, char* argv[]) {
	 // create 10 threads
		pthread_t threads[10];
		for (int i = 0; i < 10; i++)
			pthread_create(&threads[i], NULL, thread, NULL);

	 // wait for a bit then stop the threads
		sleep(3);
		for (int i = 0; i < 10; i++) {
	  // pthread_yield();  // deprecated
			sched_yield();
			pthread_cancel(threads[i]);  // cancel thread
		}
	}

I have searched through the docs to find any more comment-related options that could help me fix this, but I couldn't find any. How do I properly indent these comment lines?

**Comment by sean-mcmanus** [comment, marked as answer]
@MatejVitek It seems like it could be a clang-format bug regarding SpacesBeforeTrailingComments.

**Reply by MatejVitek** [reply]
OK, I'll head over to the clang-format page and report the issue there.
